modules/couchbase/couchbase.go

```diff
@@ -0,0 +1,597 @@
+	body := map[string]string{
+		"services": c.getEnabledServices(),
+	}
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/node/controller/setupServices", http.MethodPost, body, false)
+
+	return err
+}
+
+func (c *CouchbaseContainer) setMemoryQuotas(ctx context.Context) error {
+	body := map[string]string{}
+
+	for _, s := range c.config.enabledServices {
+		if !s.hasQuota() {
+			continue
+		}
+
+		quota := strconv.Itoa(s.minimumQuotaMb)
+		if s.identifier == kv.identifier {
+			body["memoryQuota"] = quota
+		} else {
+			body[s.identifier+"MemoryQuota"] = quota
+		}
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/pools/default", http.MethodPost, body, false)
+
+	return err
+}
+
+func (c *CouchbaseContainer) configureAdminUser(ctx context.Context) error {
+	body := map[string]string{
+		"username": c.config.username,
+		"password": c.config.password,
+		"port":     "SAME",
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/settings/web", http.MethodPost, body, false)
+
+	return err
+}
+
+func (c *CouchbaseContainer) configureExternalPorts(ctx context.Context) error {
+	host, _ := c.Host(ctx)
+	mgmt, _ := c.MappedPort(ctx, MGMT_PORT)
+	mgmtSSL, _ := c.MappedPort(ctx, MGMT_SSL_PORT)
+	body := map[string]string{
+		"hostname": host,
+		"mgmt":     mgmt.Port(),
+		"mgmtSSL":  mgmtSSL.Port(),
+	}
+
+	if contains(c.config.enabledServices, kv) {
+		kv, _ := c.MappedPort(ctx, KV_PORT)
+		kvSSL, _ := c.MappedPort(ctx, KV_SSL_PORT)
+		capi, _ := c.MappedPort(ctx, VIEW_PORT)
+		capiSSL, _ := c.MappedPort(ctx, VIEW_SSL_PORT)
+
+		body["kv"] = kv.Port()
+		body["kvSSL"] = kvSSL.Port()
+		body["capi"] = capi.Port()
+		body["capiSSL"] = capiSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, query) {
+		n1ql, _ := c.MappedPort(ctx, QUERY_PORT)
+		n1qlSSL, _ := c.MappedPort(ctx, QUERY_SSL_PORT)
+
+		body["n1ql"] = n1ql.Port()
+		body["n1qlSSL"] = n1qlSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, search) {
+		fts, _ := c.MappedPort(ctx, SEARCH_PORT)
+		ftsSSL, _ := c.MappedPort(ctx, SEARCH_SSL_PORT)
+
+		body["fts"] = fts.Port()
+		body["ftsSSL"] = ftsSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, analytics) {
+		cbas, _ := c.MappedPort(ctx, ANALYTICS_PORT)
+		cbasSSL, _ := c.MappedPort(ctx, ANALYTICS_SSL_PORT)
+
+		body["cbas"] = cbas.Port()
+		body["cbasSSL"] = cbasSSL.Port()
+	}
+
+	if contains(c.config.enabledServices, eventing) {
+		eventingAdminPort, _ := c.MappedPort(ctx, EVENTING_PORT)
+		eventingSSL, _ := c.MappedPort(ctx, EVENTING_SSL_PORT)
+
+		body["eventingAdminPort"] = eventingAdminPort.Port()
+		body["eventingSSL"] = eventingSSL.Port()
+	}
+
+	_, err := c.doHttpRequest(ctx, MGMT_PORT, "/node/controller/setupAlternateAddresses/external", http.MethodPut, body, true)
+
+	return err
+}
+
+func (c *CouchbaseContainer) configureIndexer(ctx context.Context) error {
+	storageMode := "forestdb"
+	if c.config.isEnterprise {
+		storageMode = "memory_optimized"
+	}
```

Comment:
I think it would be great to consider [this](https://github.com/testcontainers/testcontainers-java/pull/3428) in the new Go implementation

Reply:
@alihanyalcin do you think we can contribute that part? As this module will fall under the next release lifecycle, we could merge this PR as is, and you could do what @eddumelendez suggests in a follow-up. Wdyt?

Reply:
Hi, I have contributed the requested section. BTW, the enterprise edition of the Couchbase Server supports both the Plasma and Memory Optimized storage engines, while the community edition only supports the ForestDB storage engine.

Reply:
Thanks for getting so fast! We cannot be experts in all the supported technologies so we are more than thankful for seeing this kind of improvements 👏👏👏